include/fastcdr/Cdr.h

```diff
@@ -1833,14 +1934,126 @@ class Cdr
+
+            auto offset = offset_;
+
+            eprosima::fastcdr::deserialize_array(*this, array_t.data(), _Size);
+
+            if (offset_ - offset != dheader)
+            {
+                throw exception::BadParamException("Member size greater than size specified by DHEADER");
+            }
+        }
+        else
+        {
+            eprosima::fastcdr::deserialize_array(*this, array_t.data(), _Size);
+        }
+
+        return *this;
+    }
+
     /*!
      * @brief This function template deserializes a sequence of non-primitive.
      * @param vector_t The variable that will store the sequence read from the buffer.
      * @return Reference to the eprosima::fastcdr::Cdr object.
      * @exception exception::NotEnoughMemoryException This exception is thrown when trying to deserialize a position that exceeds the internal memory size.
      */
-    template<class _T, typename std::enable_if<!std::is_enum<_T>::value &&
-            !std::is_arithmetic<_T>::value>::type* = nullptr>
+    template<class _T, typename std::enable_if<
+                !std::is_enum<_T>::value &&
+                !std::is_arithmetic<_T>::value &&
+                !is_complex_array_or_string<std::vector<_T>>::value>::type* = nullptr>
+    Cdr& deserialize(
+            std::vector<_T>& vector_t)
+    {
+        uint32_t sequence_length {0};
+
+        if (CdrVersion::XCDRv2 == cdr_version_)
+        {
+            uint32_t dheader {0};
+            deserialize(dheader);
+
+            auto offset = offset_;
+
+            deserialize(sequence_length);
+
+            if (0 == sequence_length)
+            {
+                vector_t.clear();
+                return *this;
+            }
+            else
+            {
+                vector_t.resize(sequence_length);
+            }
+
+            eprosima::fastcdr::deserialize_array(*this, vector_t.data(), vector_t.size());
+
+            if (offset_ - offset != dheader)
+            {
+                throw exception::BadParamException("Member size differs from the size specified by DHEADER");
+            }
+        }
+        else
+        {
+            state state_before_error(*this);
+
+            deserialize(sequence_length);
+
+            if (sequence_length == 0)
+            {
+                vector_t.clear();
+                return *this;
+            }
+
+            if ((end_ - offset_) < sequence_length)
+            {
+                set_state(state_before_error);
+                throw exception::NotEnoughMemoryException(
+                          exception::NotEnoughMemoryException::NOT_ENOUGH_MEMORY_MESSAGE_DEFAULT);
+            }
+
+            try
+            {
+                vector_t.resize(sequence_length);
+                eprosima::fastcdr::deserialize_array(*this, vector_t.data(), vector_t.size());
+            }
+            catch (exception::Exception& ex)
+            {
+                set_state(state_before_error);
+                ex.raise();
+            }
```

Comment:
We shall follow this same part in the XCDRv2 part of the if above.

Reply:
Again, this seems to come from the old code, so a separate PR fixing it that can be backported should be done.